Add an extra nix config parameter

antifuchs opened the pull request:
I needed to add some config parameters for my personal repos (caching config, mostly), and added these clauses to my personal fork. This has worked pretty well so far, so I thought I'd upstream it.

The Readme updated perfectly with the input-documenting script, by the way (:

antifuchs committed:
Add an extra nix config parameter

samueldr commented:
Is there a trivial-enough test that could be added here to check that it works?

I guess `nix --extra-experimental-features nix-command config show` and careful grepping should be sufficient.

***If*** (I haven't looked) adding bogus options to the config is not invalid, using a name lik `lix-gha-installer-configuration-test-value = this is working` or something like that would be ideal, in that it would not interfere with other options evolving.

* * *

~~Something else to note: combining with defaults is uh... the main reason I hadn't yet done anything configurable here.~~

~~Adding to `trusted-public-keys` is “fun”, and well, the reason `sed` is used in the barbaric manner it is.~~ No, it might not be an issue..

Turns out that:

> A configuration setting usually overrides any previous value.  However, for settings that take a list of items, you can prefix the name of the setting by `extra-` to append to the previous value.

With that said, *modifying* existing values is still an issue, but that can be left to imperative commands ran after the gha installer step. So nothing to do here.

antifuchs commented:
Great idea, I'll go and add a test now!

antifuchs committed:
Add a test for nix_extra_config

This works by setting a bogus access token for example.com and then grepping for it.
